src/objects/keys.cc

```diff
@@ -733,6 +733,11 @@ Maybe<bool> KeyAccumulator::CollectInterceptorKeysInternal(
 Maybe<bool> KeyAccumulator::CollectInterceptorKeys(Handle<JSReceiver> receiver,
                                                    Handle<JSObject> object,
                                                    IndexedOrNamed type) {
+  if (v8::recordreplay::HasDivergedFromRecording()) {
+    // Disallow interceptors to execute during pauses.
+    // See: https://linear.app/replay/issue/RUN-1315#comment-26f96699
+    return Just(true);
+  }
```

Comment:
This doesn't seem like the right fix, it's pretty disconnected from the actual problem that happened in that comment and is excluding an unknown amount of keys.  Looking at the associated issue I think it would be better if we had checks in blink::CachedStorageArea::EnsureLoaded so that we're only excluding the storage info if it requires IPC with the parent process that's impossible to do while replaying.  We ran into similar problems in gecko and it's unlikely there is a long tail of issues here.

Also, when testing for whether we're doing queries on state while paused I'd like it if we shifted to testing IsReplaying() && AreEventsDisallowed() rather than HasDivergedFromRecording().  In some cases now, and more in the future, we'll want to be able to operate on the pause state while not diverged, and expect to be able to continue executing afterwards.  I did this in a few places in https://github.com/replayio/chromium/pull/445, but having a helper function for this would be better for code clarity.

Reply:
@bhackett1024 
1. I updated my comment above: Things look good. No more crashes with 150 tests run.
2. Going with the previous mantra, I decided to rather exclude keys (i.e. sacrifice accurace) than cause crashes. I guess we can slowly move toward improving accuracy as well, or at least this should be tracked with some label on the Linear issue, or a code comment. Maybe `accuracy-sacrifice`?
3. [Here](https://linear.app/replay/issue/RUN-1315#comment-26f96699) is a [link](https://source.chromium.org/search?q=%22::NamedPropertyEnumerator(%22&ss=chromium%2Fchromium%2Fsrc) that lists all such hooks in chromium: There are 7 of them. `StorageArea` is only one of them. Do you want me to manually go through and patch up all 7 of them?
4. I can change it to `IsReplaying() && AreEventsDisallowed()`, but I'd like to make a suggestion: What about a new `IsPaused` or `IsHandlingCommand` function instead?

Reply:
@Domiii I want us to avoid taking shortcuts.  What keys are being excluded here?  How will this impact the usability of the devtools?  What will happen in the future when upstream chrome starts changing the users of this API and it causes problems that are difficult to track down?  These are considerations that need to be made when making a change like this, it's not just about whether the immediate problem has been fixed.  The right answer isn't always clear cut, but in this case I think the problems with a fix whose impact is difficult to describe make it preferable to have a more targeted fix to the StorageArea code, even if we end up needing similar targeted fixes in one or two other places down the line.

Re 5 yes this is what I meant by "having a helper function for this would be better for code clarity"

Reply:
Hum, I'd prefer (1) sending this out because it fixes a lot of crashes 😅 
(2) And then track those other issues separately.
But Ok, I'll get to work.

Reply:
I fixed it [here](https://github.com/replayio/chromium/pull/459/files#diff-75654b8d30337e4d486c8eab1ce86e148cecdf50bbf224e6285c00da7d44cf20R167).
We don't want to handle this in `EnsureLoaded` because then we would produce incorrect state, such as an empty `map_` (that is not actually empty) etc.